collector/library/pod.js

```diff
@@ -162,6 +165,11 @@ class Pod {
 
         await new Promise((res, rej) => {
             request.body.on('data', async (data) => {
+                if (this.dead === true) { // Our pod has stopped, kill this request
```

Comment:
pods that are currently not receiving any data will never check for `dead`, this should be an event or ride on something else like close the incoming connection etc.